Parsing error '=>' expected when parsing typed snippets

### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.12.0

### What version of `eslint-plugin-svelte` are you using?

2.44.1

### What did you do?

```svelte
<script lang="ts">
	function hello() {
		console.log('Hello');
	}
</script>

<h1>Welcome to SvelteKit</h1>
<p>Visit <a href="https://kit.svelte.dev">kit.svelte.dev</a> to read the documentation</p>

{@render editor_button({ onclick: hello })}

<!-- prettier-ignore -->
{#snippet editor_button(foo: {
    onclick: () => void,
})}
  <button on:click={foo.onclick}>Click me</button>
{/snippet}

```


### What did you expect to happen?

I expect no errors

### What actually happened?

Received this error:
```
  12:15  error  Parsing error: '=>' expected
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/bradyisom/eslint-typed-snippets

### Additional comments

To repro with GitHub repo, `yarn install` and `yarn lint`

--- a/.changeset/short-apricots-push.md
+++ b/.changeset/short-apricots-push.md
@@ -1,0 +1,5 @@
+---
+"eslint-plugin-svelte": patch
+---
+
+fix: update svelte-eslint-parser to v0.42

--- a/packages/eslint-plugin-svelte/package.json
+++ b/packages/eslint-plugin-svelte/package.json
@@ -66,7 +66,7 @@
     "postcss-safe-parser": "^6.0.0",
     "postcss-selector-parser": "^6.1.0",
     "semver": "^7.6.2",
-    "svelte-eslint-parser": "^0.41.1"
+    "svelte-eslint-parser": "^0.42.0"
   },
   "devDependencies": {
     "@babel/core": "^7.25.8",

